common/djangoapps/student/views.py

```diff
@@ -2244,6 +2244,13 @@ def password_reset_confirm_wrapper(request, uidb36=None, token=None):
             request, uidb64=uidb64, token=token, extra_context=platform_name
         )
 
+        # If password reset was unsuccessful a template response is returned (status_code 200).
+        # Check if form is invalid then show an error to the user.
+        # Note if password reset was successful we get response redirect (status_code 302).
+        if response.status_code == 200 and not response.context_data['form'].is_valid():
```

Comment:
Yes, because we get two types of responses from the `password_reset_confirm`. 
1. HttpResponseRedirect (302): When form is valid it returns this response. Which does not have a form variable in its `context_data`.
2. TemplateResponse (200): When form is invalid. It has `context_data` and we can get form instance from it.

So I think its important to check for 200 status.